cfme/tests/storage/test_volume.py

```diff
@@ -1,54 +1,72 @@
-    # grab a volume name, the table returns a generator so use next
-
-    collection = appliance.collections.volumes
-    view = navigate_to(collection, 'All')
-
-    try:
-        volume_name = view.entities.get_first_entity().name
-    except(StopIteration, ItemNotFound):
-        pytest.skip('Skipping volume navigation for details, no volumes present')
-
-    volume = collection.instantiate(name=volume_name, provider=openstack_provider)
-
-    assert view.is_displayed
-
-    view = navigate_to(volume, 'Details')
-    assert view.is_displayed
-
-    view = navigate_to(collection, 'Add')
-    assert view.is_displayed
-
-
-@pytest.mark.uncollectif(lambda provider: not provider.one_of(OpenStackProvider))
-def test_volume_collective_crud(openstack_provider, appliance):
-    collection = appliance.collections.volumes
-    view = navigate_to(collection, 'All')
-
-    volumes = [collection.instantiate(name=item.name, provider=openstack_provider)
-              for item in view.entities.get_all()]
-
-    if volumes:
-        collection.delete(*volumes)
-    else:
-        pytest.skip("Skipping volume collective deletion, no volumes present")
+STORAGE_SIZE = 1
+
+
+def test_storage_volume_create_cancelled_validation(appliance, provider):
+    """ Test Attach instance to storage volume cancelled
+
+    prerequisites:
+        * Storage provider
+
+    Steps:
+        * Navigate to storage add volume page
+        * Click Cancel button
+        * Assert flash message
+    """
+    volume_collection = appliance.collections.volumes
+    manager_name = '{} Cinder Manager'.format(provider.name)
+    volume_collection.create(name=fauxfactory.gen_alpha(),
+                             storage_manager=manager_name,
+                             tenant=provider.data['provisioning']['cloud_tenant'],
+                             size=STORAGE_SIZE,
+                             provider=provider,
+                             cancel=True)
+
+    view = volume_collection.create_view(VolumeAllView)
+    view.flash.assert_message('Add of new Cloud Volume was cancelled by the user')
+
+
+@pytest.mark.tier(1)
+def test_storage_volume_crud(appliance, provider):
+    """ Test storage volume crud
+
+    prerequisites:
+        * Storage provider
+
+    Steps:
+        * Crate new volume
+        * Update volume name
+        * Delete volume
+    """
+    # create volume
+    volume_collection = appliance.collections.volumes
+    manager_name = '{} Cinder Manager'.format(provider.name)
+    volume = volume_collection.create(name=fauxfactory.gen_alpha(),
+                                      storage_manager=manager_name,
+                                      tenant=provider.data['provisioning']['cloud_tenant'],
+                                      size=STORAGE_SIZE,
+                                      provider=provider)
+    assert volume.exists
+
+    # update volume
+    new_volume_name = fauxfactory.gen_alpha()
+    volume.edit(new_volume_name)
```

Comment:
**REQUIRED**
Please use a context manager here:
```python
with update(volume):
    volume.name = new_volume_name
```
`Volume` should inherit `Updateable` mixin and implement `update` method, not `edit`.

Reply:
@quarckster added update method